core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -325,6 +346,13 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    */
   private def roundDownToExactMultiple(number: Int, factor: Int) = factor * (number / factor)
 
+  private def toRelative(offset: Long): Option[Int] = {
+    val relativeOffset = offset - baseOffset
+    if (relativeOffset < 0 || relativeOffset > Int.MaxValue)
+      return None
```

Comment:
nit: explicit returns are frowned upon in scala. `if` expressions in scalas are values, so we would tend to prefer the following:

```scala
if (relativeOffset < 0 || relativeOffset > Int.MaxValue)
  None
else
  Some(relativeOffset.toInt)
```